src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/Container.h

```diff
@@ -0,0 +1,249 @@
+    */
+    void trimRight(double cutoff);
+
+    /** @brief Trims the left side of the isotope distribution to isotopes with a significant contribution.
+
+            If the isotope distribution is calculated for large masses (and with high max isotope)
+            it might happen that many entries contain only small numbers. This function can be
+            used to remove these entries.
+
+            Do consider normalising the distribution afterwards.
+    */
+    void trimLeft(double cutoff);
+
+
+    
+    bool isNormalized() const;
+
+
+    bool isConvolutionUnit() const;
+    //@}
+
+    /// @name Operators
+    //@{
+    /// Assignment operator
+    IsotopeDistribution & operator=(const IsotopeDistribution & isotope_distribution);
+
+    /// equality operator, returns true if the @p isotope_distribution is identical to this, false else
+    bool operator==(const IsotopeDistribution & isotope_distribution) const;
+
+    /// inequality operator, returns true if the @p isotope_distribution differs from this, false else
+    bool operator!=(const IsotopeDistribution & isotope_distribution) const;
+    //@}
+
+    /// @name Iterators
+    //@{
+    inline Iterator begin() { return distribution_.begin(); }
+
+    inline Iterator end()   { return distribution_.end(); }
+
+    inline ConstIterator begin() const { return distribution_.begin(); }
+
+    inline ConstIterator end() const { return distribution_.end(); }
+
+    inline ReverseIterator rbegin() { return distribution_.rbegin(); }
+
+    inline ReverseIterator rend()   { return distribution_.rend(); }
+
+    inline ConstReverseIterator rbegin() const { return distribution_.rbegin(); }
+
+    inline ConstReverseIterator rend() const { return distribution_.rend(); }
+
+    inline void insert(const Peak1D::CoordinateType& mass, const Peak1D::IntensityType& intensity)
+    {
+      distribution_.push_back(Peak1D(mass, intensity));
+    }
+    //@}
+
+    /// @name Convolutional Dummy Operators
+    //@{
+    /// operator which adds this distribution and the @p isotope_distribution to return IsotopeDisribution (similar to convolve distributions)
+    IsotopeDistribution operator+(const IsotopeDistribution & isotope_distribution) const;
+
+    /// operator which adds @p isotope_distribution to this (similar to convolve distributions)
+    IsotopeDistribution & operator+=(const IsotopeDistribution & isotope_distribution);
+
+    /// operator which multiplies this distribution by @p factor (similar to @p factor times applying operator '+')
+    IsotopeDistribution operator*(Size factor) const;
+
+    /// operator which multiplies this distribution by @p factor (similar to @p factor times applying operator '+=')
+    IsotopeDistribution & operator*=(Size factor);
+    //@}
+
+
+    /// @name Data Access Operators
+    //@{
+    /// operator which access a cell of the distribution and wraps it in SpectrumFragment struct
+    Peak1D& operator[](const UInt64& index){ return distribution_[index];}
```

Comment:
please, never ever inherit from STL classes in a library. We've had our share of trouble with it (we tried, and failed badly in many ways).